File: Domain/Models/Inventory/Supplier.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.Models.Inventory
{
    public class Supplier
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required, StringLength(250)]
        public string Name { get; set; } = string.Empty;

        [StringLength(50)]
        public string? Phone { get; set; }

        [StringLength(100)]
        public string? Email { get; set; }

        [StringLength(500)]
        public string? Address { get; set; }

        // Egyptian tax registration number
        [StringLength(50)]
        public string? TaxRegistrationNumber { get; set; }

        // Egyptian commercial registration
        [StringLength(50)]
        public string? CommercialRegister { get; set; }

        public decimal Balance { get; set; }

        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<PurchaseInvoice>? PurchaseInvoices { get; set; }
    }
}

File: Mobile/lib/features/inventory/models/supplier.dart
class Supplier {
  final String id;
  final String name;
  final String? phone;
  final String? email;
  final String? address;
  final String? taxRegistrationNumber;
  final double balance;
  final bool isActive;

  Supplier({
    required this.id,
    required this.name,
    this.phone,
    this.email,
    this.address,
    this.taxRegistrationNumber,
    required this.balance,
    required this.isActive,
  });

  factory Supplier.fromJson(Map<String, dynamic> j) => Supplier(
        id: j['id'],
        name: j['name'] ?? '',
        phone: j['phone'],
        email: j['email'],
        address: j['address'],
        taxRegistrationNumber: j['taxRegistrationNumber'],
        balance: (j['balance'] ?? 0).toDouble(),
        isActive: j['isActive'] ?? true,
      );
}
